src/sql/mod.rs

```diff
@@ -12,34 +12,41 @@ pub(crate) mod escape;
 mod ser;
 
 #[derive(Clone)]
-pub(crate) enum SqlBuilder {
-    InProgress(Vec<Part>),
+pub(crate) enum SqlBuilder<'a> {
+    InProgress(Vec<Part<'a>>),
     Failed(String),
 }
 
 #[derive(Clone)]
-pub(crate) enum Part {
+pub(crate) enum Part<'a> {
     Arg,
     Fields,
-    Text(String),
+    Str(&'a str),
+    String(String),
 }
 
-impl SqlBuilder {
-    pub(crate) fn new(template: &str) -> Self {
-        let mut iter = template.split('?');
-        let prefix = String::from(iter.next().unwrap());
-        let mut parts = vec![Part::Text(prefix)];
-
-        for s in iter {
-            let text = if let Some(text) = s.strip_prefix("fields") {
+impl<'parts> SqlBuilder<'parts> {
+    pub(crate) fn new(template: &'parts str) -> Self {
+        let mut parts = Vec::new();
+        let mut rest = template;
+        while let Some(idx) = rest.find('?') {
+            if idx != 0 {
+                parts.push(Part::Str(&rest[..idx]));
+            }
+            rest = &rest[idx + 1..];
+            if let Some(restqq) = rest.strip_prefix('?') {
+                parts.push(Part::Str("?"));
```

Comment:
Instead of being a separate Part, this could have the above part slice to idx+1 instead of idx